jax/interpreters/pxla.py

```diff
@@ -479,9 +480,9 @@ def __init__(self,
     # TODO(skye): assert invariants. Keep performance in mind though.
     if indices is None:
       indices = spec_to_indices(aval.shape, sharding_spec)
-    self.aval = aval
-    self.device_buffers = device_buffers
-    self.sharding_spec = sharding_spec
+
+    pmap_lib.ShardedDeviceArray.__init__(self, aval, sharding_spec,
+                                         device_buffers)
```

Comment:
Also, `super().__init__`?

Reply:
I renamed the CL to say it is also changing ShardedDEviceArray, which now extends a C++ type.

It's done by:
- extending the C++ object.
- it will serve aval, sharding_spec and device_buffers, which are removed from __slots__
- calling this contructor.